src/unicode.c

```diff
@@ -737,7 +737,7 @@ int uoffset(AL_CONST char *s, int index)
       }
    }
 
-   return (long)s - (long)orig;
+   return (long long)s - (long long)orig;

```

Comment:
The cast is pointless to begin with, just change it to: "return s - orig;" instead